### PR TITLE
#597 json multiline: added a test for multiline json values

### LINKED ISSUE #597: Test the JSON variable text value Step with multi-line text
`(?:the )?text(?: in)?(?: the)?(?: JSON)?(?: var)?(?:iable)?(?: )?"([^"]+)" should be` (https://github.com/SuffolkLITLab/ALKiln/blob/releases/v4/lib/steps.js#L656)

We should test that the above Step can properly test a multi-line string. From testing in the wild, this does seem to work, so it's not urgent, but it would be good to have it under our belts.

### PR DESCRIPTION
**Pull request by jostran14**
specifically for variables set by code not through user input.
closes #597

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -40,6 +40,7 @@ Format:
 -->
 ## [Unreleased]
 ### Added
+- added a test for multiline json values (only ones set by code not by user input)
```
Comment (BryceStevenWilley) [inline CHANGELOG.md]: The "added" section usually has the specific meaning of "something added to the program interface", i.e. you as the user of this program can do something you couldn't before. This PR probably deserves it's own section, either "Internal" or "Testing", if it needs to be in the changelog at all.  Defering to @plocket, the changelog expert.

File: docassemble/ALKilnTests/data/sources/observation_steps.feature [hunk 2]
```diff
@@ -99,23 +99,26 @@ Scenario: Test "Then I don’t continue" with an apostrophe
   And I will be told an answer is invalid
 
 @fast @o9 @table @json
-Scenario: I can match JSON page var to str
```
Comment (BryceStevenWilley) [inline docassemble/ALKilnTests/data/sources/observation_steps.feature:102]: I'm not sure why this scenario was removed, it should likely still be around?

File: docassemble/ALKilnTests/data/sources/observation_steps.feature [hunk 3]
```diff
@@ -99,23 +99,26 @@ Scenario: Test "Then I don’t continue" with an apostrophe
...
+  """
+  This block is.
+  A multiline block.
+  """
+  Then the text in the JSON variable "single_line_val" should be
   """
-  dropdown_opt_2
+  This is one line.
   """
+  #Then the text in the JSON variable "number_val" should be
```
Comment (BryceStevenWilley) [inline docassemble/ALKilnTests/data/sources/observation_steps.feature]: From the comments in the issue, it looks like checking the number wasn't working properly. I'd remove the commented lines here and the `number_val` variable from the yaml interview if we don't want to try to make it work in this PR.

**Review by BryceStevenWilley — COMMENTED** [review]
A good PR! The few suggestions that I'd recommend before merging are in comments

File: CHANGELOG.md (hunk 1 above)
Comment (plocket) [reply CHANGELOG.md]: I like the idea of adding `### Internal` to our headings! Thanks!

Comment (jostran14) [reply CHANGELOG.md]: Thanks for the input, and we discussed and implemented an internal section to the change log.

File: docassemble/ALKilnTests/data/sources/observation_steps.feature (hunk 2 above)
Comment (jostran14) [reply docassemble/ALKilnTests/data/sources/observation_steps.feature:102]: @plocket and I considered these 2 tests equivalent. The purpose of this test is to make sure ALKiln can match a variable in the json to a value that the test author expects. We concluded that it shouldn't matter how that json variable is set. The previous version of this test used a variable set by filling in a field. The current version uses a variable set by python. We couldn't think of a reason that those 2 methods are testing anything different, and we didn't want duplicate tests. Do you still think we should restore the previous test?

Comment (BryceStevenWilley) [reply docassemble/ALKilnTests/data/sources/observation_steps.feature:102]: Eh, yeah, I'd lean towards leaving it. I looked around, and surprisingly, there aren't any tests that seem to actually verify the value of a selection made by the user; we just say to input it and continue on. You're right that it's redundant for testing our JSON stuff, but we also need to test docassemble itself, which that test does.

Comment (plocket) [reply docassemble/ALKilnTests/data/sources/observation_steps.feature:102]: We can add it back in, but if we're going to test selections, etc, I think that test belongs in its own set of tests and not buried in a JSON Step test. Again, we can still add this back in here for the moment and just make an issue to move it later and add further tests, it just doesn't seem like this is the appropriate place for it. I'm not sure what you mean by testing docassemble itself and would love to talk more about that if we get a chance. I those tests, too, would belong in a different set of tests.

**Comment by plocket** [comment]
Just need to make the issue 👍

File: docassemble/ALKilnTests/data/sources/observation_steps.feature (hunk 2 above)
Comment (jostran14) [reply docassemble/ALKilnTests/data/sources/observation_steps.feature:102]: https://github.com/SuffolkLITLab/ALKiln/issues/742

**Review by plocket — APPROVED** [review]
Looks great! As soon as the tests pass, this can be merged!

**Pull request merged**